test/core/error/token/test.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA
+ *
+ */
+
+define(['core/error/types', 'core/error/TokenError'], function (errorTypes, TokenError) {
+    'use strict';
+
+    QUnit.module('TokenError');
+
+    QUnit.test('construct', assert => {
+        assert.expect(6);
+
+        const err = new TokenError('Token expired', null);
+        assert.equal(err.name, 'TokenError');
+        assert.equal(err.type, errorTypes.token);
+        assert.equal(err.message, 'Token expired');
+        assert.equal(err.errorCode, 401);
+        assert.equal(err.response, null);
+        assert.equal(err.recoverable, false);
```

Comment:
Maybe these parameters should be checked when their values are defined. This test tests only the default values.

Reply:
Firstly i rethink a little income parameters for new error type and modify specification. I've added new test which check full list of handled params